docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -11,28 +11,82 @@ This guide helps you to utilize indexes on mutable dataset.
-removing an existing one, required a full refresh to make the index usable again. This could be
-a costly operation due to shuffling and repartitioning all records in the latest source data.
+Previously, any change in the original dataset content required a full refresh to make the index usable again.
+This could be a costly operation due to shuffling and repartitioning all records in the latest source data.
 
-Now, we offer several options to handle appended and deleted files more efficiently.
+Now, we offer several options to handle above scenario more efficiently.
 
 ## Options of using index when your dataset changes
 
 ### Refresh Index
+You can refresh an index according to its latest source data files by running the `refreshIndex` command.
+Hyperspace provide several modes to refresh an index. These modes differ in terms of the way they update the index and the amount of data scans and shuffle each does.
+You should pick a mode for refreshing an index according to its current size and total amount of data deleted from or appended to its source data files.
+You can specify the mode as an argument when calling the `refreshIndex` command.
+Currently, there are two refresh modes available for an index: `"full"` and `"incremental"`.
 
-We provide several refresh modes to refresh index data.
+#### Refresh Index - Full Mode
+After some changes in an index's original dataset files, using `refreshIndex` with the `"full"` mode causes
+the index refresh action perform a full rebuild of the index.
+This ends up creating a new version of the index and involves a full scan and shuffle of its latest source data.
+As a result, the amount of time it takes to refresh an index in this mode is similar to creating a new index, with the same index configuration, on the latest source data.
+The advantage of using the full mode is that once index refresh is finished successfully, new index files are organized in the most optimized way according to index's latest source data content and its bucketing configuration.
 
-#### Full
-TODO
+Assume you have an index with the name `empIndex`. After adding and removing some data files from the dataset `empIndex` is created on, you can refresh it in the full mode as below:
 
-#### Incremental
-TODO
+Scala:
+```scala
+import com.microsoft.hyperspace._
+
+val hs = new Hyperspace(spark)
+hs.refreshIndex("empIndex", "full")
+``` 
+
+Python:
+
+```python
+from hyperspace import Hyperspace
+
+hs = Hyperspace(spark)
+hs.refreshIndex("empIndex", "full")
+```
+
+#### Refresh Index - Incremental Mode
+After some files are added to or deleted from the original source files, an index was built on,
+using `refreshIndex` with the `"incremental"` mode causes the index refresh action recreate any existing index file,
+which has some deleted records, to remove those records. Refresh also creates new index files by indexing newly added data files.
+An index needs to have lineage enabled to be eligible for refresh in the incremental mode.
```

Comment:
@pirz @imback82 @apoorvedave1 Append-only dataset doesn't need lineage column for "incremental" mode. 
As using linage column has some overhead, I think we need describe this in user document, in a clear and easy way..

Reply:
@pirz Feel free to update this in #217. Thanks!

Reply:
@sezruby @imback82 - Just to be clear, for supporting deletes, index needs lineage. Are you asking to modify above sentence as below?
"An index needs to have lineage enabled to support delete during refresh in the incremental mode."

Reply:
@imback82 @sezruby - I modified and extended this part in #217 - Please take a look at it there and let me know about your comments (if any). Thanks!

Reply:
> Just to be clear, for supporting deletes, index needs lineage. Are you asking to modify above sentence as below?
"An index needs to have lineage enabled to support delete during refresh in the incremental mode."

Yes I'd like to point out that lineage is only required for deleted files, so something like:
In order to support deleted files in the incremental refresh mode, lineage should be enabled when it is built. In case it is append-only dataset, (only appended files are expected in your dataset), you do not need to set lineage column.